packages/cashier/src/stores/payment-agent-store.ts

```diff
@@ -1,39 +1,54 @@
-import { action, computed, observable } from 'mobx';
+import { action, computed, observable, IObservableArray } from 'mobx';
+import { PaymentAgentDetailsResponse } from '@deriv/api-types';
 import { formatMoney, routes } from '@deriv/shared';
 import Constants from 'Constants/constants';
 import ErrorStore from './error-store';
 import VerificationStore from './verification-store';
+import {
+    TAgent,
+    TExtendedPaymentAgentList,
+    TExtendedPaymentAgentListResponse,
+    TPartialPaymentAgentList,
+    TPaymentAgentWithdrawConfirm,
+    TPaymentAgentWithdrawRequest,
+    TPaymentAgentWithdrawReceipt,
+    TRootStore,
+    TSupportedBank,
+    TTarget,
+    TWebSocket,
+} from 'Types';
 
 export default class PaymentAgentStore {
-    constructor({ WS, root_store }) {
+    constructor(public WS: TWebSocket, public root_store: TRootStore) {
         this.root_store = root_store;
         this.WS = WS;
     }
 
-    @observable list = [];
-    @observable agents = [];
+    @observable list: IObservableArray<TPartialPaymentAgentList> = [];
+    @observable agents: IObservableArray<TAgent> = [];
     @observable container = Constants.containers.payment_agent;
-    @observable error = new ErrorStore();
-    @observable filtered_list = [];
+    @observable error: TRootStore['modules']['cashier']['error'] = new ErrorStore();
+    @observable filtered_list: IObservableArray<TPartialPaymentAgentList> = [];
     @observable is_name_selected = true;
     @observable is_withdraw = false;
     @observable is_try_withdraw_successful = false;
     @observable is_withdraw_successful = false;
-    @observable confirm = {};
-    @observable receipt = {};
-    @observable selected_bank = 0;
-    @observable supported_banks = [];
+    @observable confirm: TPaymentAgentWithdrawConfirm = {};
+    @observable receipt: TPaymentAgentWithdrawReceipt = {};
+    @observable selected_bank: number | string = 0;
+    @observable supported_banks: IObservableArray<TSupportedBank> = [];
     @observable verification = new VerificationStore({ root_store: this.root_store, WS: this.WS });
     @observable active_tab_index = 0;
-    @observable all_payment_agent_list = [];
```

Comment:
fix `all_payment_agent_list ` variable, because it was not an array, it is PaymentAgentListResponse : {
 paymentagent_list?: PaymentagentList;
  echo_req: {
    [k: string]: unknown;
  };
  msg_type: "paymentagent_list";
  req_id?: number;
  [k: string]: unknown;
}